src/axom/core/memory_management.hpp

```diff
@@ -110,6 +110,36 @@ inline int getDefaultAllocatorID()
 #endif
 }
 
+/*!
+ * \brief Returns the ID of the allocator that allocated the memory pointed
+ *        to by \a ptr.
+ * \param ptr A pointer to memory.
+ * \return ID of the allocator that allocated the memory.
+ */
+/// &{
+#ifdef AXOM_USE_UMPIRE
+inline int getAllocatorIDForAddress(void* ptr)
+{
+  umpire::ResourceManager& rm = umpire::ResourceManager::getInstance();
+  return rm.getAllocator(ptr).getId();
```

Comment:
What if `ptr` was not Umpire-allocated?  A check that the allocator could be found would be helpful to avoid crashing.  Would it make sense to return `INVALID_ALLOCATOR_ID` in that case?

Reply:
Good suggestion. I added a test that calls `getAllocatorIDForAddress()` with a pointer not allocated by Umpire, which made Umpire throw an exception. I added exception handling and return INVALID_ALLOCATOR_ID in that case.